### PR TITLE
iFixit API: have a little more grace

### PR DESCRIPTION
**Pull request by danielbeardsley**
This is a fast API (10-20ms), yet we still see occasional timeouts.
Just retry in this case, cause if the original request took longer
than 3s, it'll probably never finish and must have got lost in the
shuffle.

Closes https://github.com/iFixit/ifixit/issues/44076

### CODE REVIEW

File: src/iFixit/EventListener/iFixitApi.php [hunk 1]
```diff
@@ -5,6 +5,7 @@
 use iFixit\Akeneo\iFixitBundle\iFixitConfig;
 
 use GuzzleHttp\Psr7\Request;
+use GuzzleHttp\{Client, HandlerStack, Middleware, RetryMiddleware};
```
Comment (davidrans) [inline src/iFixit/EventListener/iFixitApi.php]:
```suggestion
use GuzzleHttp\{Client, HandlerStack, Middleware};
```

**Review by davidrans — CHANGES_REQUESTED** [review]
CR :coffee:

**Review by davidrans — APPROVED** [review]

**Review by davidrans — APPROVED** [review]
CR :coffee:

**Comment by k0rvusk0r4x** [comment]
Not sure how to go about testing this. Basically, just want to make sure it's not going to bog anything down with retries to the point it somehow make things worse overall. (I don't really see that happening with an API that is naturally fast, though.)

**Comment by k0rvusk0r4x** [comment]
Thoughts on how to test [this], @danielbeardsley?

**Comment by danielbeardsley** [comment]
QA:

1. From dev, run `dev-akeneo` to SSH into the akeneo machine
2. `cd /opt/akeneo`
3. fetch and checkout this branch
4. `make down && make up`
5. Click "save" on a product in Akeneo
6. Ensure product data has updated by visiting the product page (OR tail the debug log in `/home/ifixit/Code`)

**Comment by k0rvusk0r4x** [comment]
Followed steps one through four [here](https://github.com/iFixit/pim-community-standard/pull/20#issuecomment-1238398178).
When I went to akeneo.cominor.com, I got this:
image

dev_block 😿

File: src/iFixit/EventListener/iFixitApi.php [hunk 2]
```diff
@@ -25,12 +26,18 @@ public function __construct(iFixitConfig $config, LoggerInterface $logger) {
       $this->config = $config;
       $this->logger = $logger;
 
+      $stack = HandlerStack::create();
+      $stack->push(Middleware::retry(function ($retries, $request, $response = null) {
+         return $retries <= 2;
+      });
```
Comment (danielbeardsley) [inline src/iFixit/EventListener/iFixitApi.php]:
```suggestion
      }));
```

I guess we don't have a syntax check for CI on this repo ;-)

**Comment by danielbeardsley** [comment]
un_dev_block :+1: 
 
Derp, thanks @k0rvusk0r4x !

**Review by mlahargou — APPROVED** [review]
CR 📱

**Comment by deltuh-vee** [comment]
QA 🎬 
Akeneo product updates still work with this change.

**Pull request merged**